public/script.js

```diff
@@ -5704,15 +5704,26 @@ function extractMessageFromData(data) {
  * @returns {string} Extracted reasoning
  */
 function extractReasoningFromData(data) {
-    if (main_api === 'openai' && oai_settings.show_thoughts) {
-        switch (oai_settings.chat_completion_source) {
-            case chat_completion_sources.DEEPSEEK:
-                return data?.choices?.[0]?.message?.reasoning_content ?? '';
-            case chat_completion_sources.OPENROUTER:
-                return data?.choices?.[0]?.message?.reasoning ?? '';
-            case chat_completion_sources.MAKERSUITE:
-                return data?.responseContent?.parts?.filter(part => part.thought)?.map(part => part.text)?.join('\n\n') ?? '';
-        }
+    switch (main_api) {
+        case 'textgenerationwebui':
+            switch (textgen_settings.type) {
+                case textgen_types.OPENROUTER:
+                    return data?.choices?.[0]?.reasoning ?? '';
```

Comment:
Shouldn't TC have a similar flag - or respect the same internal flag?
Because including models thoughts should be a user opt-in, and not on by default.

You even have the Openrouter checkbox, but are not checking it here, right?

Reply:
OpenRouter doesn't output `reasoning` contents when `include_reasoning` is false. So this will be the empty string when reasoning is not requested.